### PR TITLE
Added support for mifareclassic and mifare ultralight cards

### PR DESCRIPTION
**Pull request by tblanchard**
Added support to read MiFareClassic and Mifare Ultralight cards.  I'm cleaning up my repositories, I will leave this open for a couple weeks, then I'm deleting it as I have no reason to keep it going forward

### CODE REVIEW

File: src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java [hunk 1]
```diff
@@ -622,10 +650,14 @@ public void run() {
                     for (String tagTech : tag.getTechList()) {
                         Log.d(TAG, tagTech);
                         if (tagTech.equals(NdefFormatable.class.getName())) {
-                            fireNdefFormatableEvent(tag);
+                            //fireNdefFormatableEvent(tag);
```
Comment (mobidev111) [inline src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java]: what's the reason for this being commented out? either keep, remove or add a comment on why it is commented out

File: src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java [hunk 2]
```diff
@@ -650,9 +682,169 @@ private void fireNdefEvent(String type, Ndef ndef, Parcelable[] messages) {
...
+            protected void onPostExecute(String command)
+            {
+                webView.sendJavascript(command);
+            }
+        }.execute(tag);
+    }
+
+    private void fireNdefFormatableEvent (Tag tag) {
+        JSONObject tagJSON = Util.tagToJSON(tag);
+        /*
```
Comment (mobidev111) [inline src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java]: what's the reason for this being commented out? either keep, remove or add a comment on why it is commented out

File: src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java [hunk 3]
```diff
@@ -650,9 +682,169 @@ private void fireNdefEvent(String type, Ndef ndef, Parcelable[] messages) {
...
+                    mifare.connect();
+                    mifare.setTimeout(3000);
+
+                    for(int i = 0; i < mifare.getBlockCount(); ++i)
+                    {
+                        if(mifare.blockToSector(i) != sector)
+                        {
+                            sector = mifare.blockToSector(i);
+                            mifare.authenticateSectorWithKeyA(sector,mifare.KEY_DEFAULT);
+                            //mifare.authenticateSectorWithKeyB(sector,mifare.KEY_DEFAULT);
```
Comment (mobidev111) [inline src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java]: what's the reason for this being commented out? either keep, remove or add a comment on why it is commented out

**Review by mobidev111 — COMMENTED** [review]
would love to see this pulled in. some stray comments in there - see review

**Comment by IllusionVK** [comment]
@tblanchard Just wondering if you would be able to help out with @mobidev111 questions ?

It would be really great to get these additions merged.  **Appreciate your time is limited and so thanks in advance for any help you can provide.**

**Pull request closed without merging**